Neos.Utility.ObjectHandling/Classes/ObjectAccess.php

```diff
@@ -435,45 +435,13 @@ public static function buildSetterMethodName(string $propertyName): string
      * Instantiates the class named `$className` using the `$arguments` as constructor
      * arguments (in array order).
      *
-     * For less than 7 arguments `new` is used, for more a `ReflectionClass` is created
-     * and `newInstanceArgs` is used.
-     *
-     * Note: this should be used sparingly, just calling `new` yourself or using Dependency
-     * Injection are most probably better alternatives.
-     *
      * @param string $className
      * @param array $arguments
      * @return object
+     * @deprecated directly use "new $className(...$arguments)" instead
```

Comment:
We could postpone the deprecation to 5.2 if wanted. Does it hurt though?

Reply:
nah, let's deprecatae this right away. We can always decide to leave it in for another round of major...